src/rust/ide/enso-protocol/src/language_server/types.rs

```diff
@@ -29,6 +29,18 @@ pub struct Path {
 
 }
 
+impl From<&FileSystemObject> for Path {
+    fn from(file_system_object:&FileSystemObject) -> Path {
+        match file_system_object {
+            FileSystemObject::Directory{name,path}          => path.clone_with_segment(name),
```

Comment:
The name `clone_with_segment` is not understnadable. It is `append` like? Immutable one? If `Path` is designed to be always immutable, just name it `append`. If `Path` is designed to be mutable, name it `append_im` (like a lot of Rust libs do). I believe that `Path` should be immutable by design btw.